### PR TITLE
Add offices

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#1392 /bcgov/entity#1544

*Description of changes:*

Implemented New UI for Addresses displayed on dashboard.

Handling new addresses response from API which includes a registered office and records office.

Added some typescript refactoring.

Updated Unit testing.

* This PR does not include the code refactor for the change of address filing itself, only the dashboard. The next pr will include change of address filing and was done to make the PR more manageable. 

* Dependencies

This PR contains the rebased changes from bcgov/master and will appear very large until the story branch is in sync with master.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by katiemcgoff** [comment]
I've created a new story branch (because I screwed up the original one) so this one is master with Peter's previous office address changes. There are a few commits in your branch that might not actually be yours but they're probably few enough that we can manage the conflicts.

There are a few existing conflicts, including a migration that needs to be reconciled - down_revision should be the last one in the new chain.

Also note that the schema tests were simplified - the simple positive tests (test AR, test COA, etc) are grouped in the test_valid_filing() test, and you can add to it with the list of example_data/schema_data - ALL_FILINGS.

**Comment by cameron-freshworks** [comment]
> I've created a new story branch (because I screwed up the original one) so this one is master with Peter's previous office address changes. There are a few commits in your branch that might not actually be yours but they're probably few enough that we can manage the conflicts.
> 
> There are a few existing conflicts, including a migration that needs to be reconciled - down_revision should be the last one in the new chain.
> 
> Also note that the schema tests were simplified - the simple positive tests (test AR, test COA, etc) are grouped in the test_valid_filing() test, and you can add to it with the list of example_data/schema_data - ALL_FILINGS.

Excellent, thanks Katie. I think Peter mentioned he will have a look at this soon as he can :)

File: coops-ui/src/App.vue [hunk 1]
```diff
@@ -117,18 +117,18 @@ export default {
         Promise.all([
           this.getBusinessInfo(businessId),
           axios.get(businessId),
-          axios.get(businessId + '/tasks'),
-          axios.get(businessId + '/filings'),
+          // axios.get(businessId + '/tasks'),
```
Comment (severinbeauvais) [inline coops-ui/src/App.vue]: If this was just for testing, please ensure you restore this code before it gets merged!

File: coops-ui/src/interfaces/address-interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import Vue from 'vue'
+
+export interface BaseAddressType extends Vue {
+  $refs: any
+}
+
+export interface AddressIF extends Vue {
```
Comment (severinbeauvais) [inline coops-ui/src/interfaces/address-interfaces.ts]: Awesome, I love this!
If this should match the address schema then please add a suitable comment. (For example, in some cases, region can be null.)

File: coops-ui/src/App.vue (hunk 1 above)
Comment (cameron-freshworks) [reply coops-ui/src/App.vue]: Absolutely, thanks for reminder. There was a pending update on the backend with regards to these filings and the seed file. Should be good to uncomment!

File: coops-ui/src/interfaces/address-interfaces.ts (hunk 2 above)
Comment (cameron-freshworks) [reply coops-ui/src/interfaces/address-interfaces.ts]: Cool, thanks!   Okie doke, sounds great, i will look into that!

**Pull request merged**